resolve.c

```diff
@@ -108,11 +116,20 @@ void resolveForwardDestinations(bool onlynew)
 		if(onlynew && !forwarded[forwardID].new)
 			continue;
 
+		// Lock data when obtaining IP of this forward destination
 		lock_shm();
+		const char* ipaddr = getstr(forwarded[forwardID].ippos);
+		unlock_shm();
 
-		forwarded[forwardID].namepos = addstr(resolveHostname(getstr(forwarded[forwardID].ippos)));
```

Comment:
See above comment for the client host name resolution. This one-lines is affected by the same problem.